ClinVar Phase 1

toneillbroad opened the pull request:
Tables, assets and generated Jade schema for review.

toneillbroad committed:
Added tables, assets and ceated the clinvar_schema.json for ClinVar phase 1.

danxmoran commented on schemas/clinvar/assets/ClinicalAssertion.json:
```diff
@@ -0,0 +1,9 @@
+    "name": "ClinicalAssertion",
+    "rootTable": "clinical_assertion",
+    "rootColumn": "clinical_assertion_id",
+    "tables": [
```
I think we can leave out the `table` and `follow` fields from the checked-in asset definitions

danxmoran commented on schemas/clinvar/clinvar_schema.json:
```diff
@@ -0,0 +1,1736 @@
+{
```
I'm not sure if we need to check in the full schema, since it's generated.

danxmoran commented on schemas/clinvar/tables/ClinicalAssertion.json:
```diff
@@ -0,0 +1,103 @@
+    "name": "clinical_assertion",
+    "columns": [
+	{
+	    "name": "clinical_assertion_id",
```
Here and elsewhere, I think it'd help clarity to name the primary ID column `id` in each table, instead of having a variable name. That'll make it easy to distinguish primary IDs from foreign ID references when scanning the column names.
```suggestion
	    "name": "id",
```

toneillbroad replied:
Agree.

danxmoran commented on schemas/clinvar/tables/ClinicalAssertion.json:
```diff
@@ -0,0 +1,103 @@
+	},
+	{
+	    "name": "date_created",
+	    "datatype": "date"
```
Are these just dates, or are they actually timestamps?

toneillbroad replied:
All dates are in the form YYYY-MM-DD.

danxmoran commented on schemas/clinvar/tables/ClinicalAssertion.json:
```diff
@@ -0,0 +1,103 @@
+	},
+	{
+	    "name": "variation_archive_variation_id",
+	    "datatype": "int"
```
I think this needs to be either `integer` or `int64`. We've been using `integer` elsewhere.
```suggestion
	    "datatype": "integer"
```

toneillbroad replied:
I'll switch all of the ints to integer.

danxmoran commented on schemas/clinvar/tables/ClinicalAssertion.json:
```diff
@@ -0,0 +1,103 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "rcv_accession_id",
```
Should this have `links`?

danxmoran commented on schemas/clinvar/tables/ClinicalAssertion.json:
```diff
@@ -0,0 +1,103 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "variation_archive_id",
```
Should this have `links`?

danxmoran commented on schemas/clinvar/tables/ClinicalAssertion.json:
```diff
@@ -0,0 +1,103 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "variation_archive_variation_id",
```
Should this have `links`?

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTrait.json:
```diff
@@ -0,0 +1,37 @@
+    "name": "clinical_assertion_trait",
+    "columns": [
+	{
+	    "name": "clinical_assertion_trait_id",
```
```suggestion
	    "name": "id",
```

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTrait.json:
```diff
@@ -0,0 +1,37 @@
+    "columns": [
+	{
+	    "name": "clinical_assertion_trait_id",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTrait.json:
```diff
@@ -0,0 +1,37 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "xrefDB",
```
Snake-case for convention.
```suggestion
	    "name": "xref_db",
```

toneillbroad replied:
Good find.

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTrait.json:
```diff
@@ -0,0 +1,37 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "xrefID",
```
```suggestion
	    "name": "xref_id",
```

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTrait.json:
```diff
@@ -0,0 +1,37 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "clinical_assertion_trait_set_id",
```
Should this have `links`?

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTrait.json:
```diff
@@ -0,0 +1,37 @@
+	    "datatype": "int"
+	},
+	{
+	    "name": "variation_archive_trait_id",
```
Should this have `links`?

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTrait.json:
```diff
@@ -0,0 +1,37 @@
+	},
+	{
+	    "name": "clinical_assertion_trait_set_id",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTrait.json:
```diff
@@ -0,0 +1,37 @@
+	},
+	{
+	    "name": "variation_archive_trait_id",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTraitSet.json:
```diff
@@ -0,0 +1,31 @@
+    "name": "clinical_assertion_trait_set",
+    "columns": [
+	{
+	    "name": "clinical_assertion_trait_set_id",
```
```suggestion
	    "name": "id",
```

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTraitSet.json:
```diff
@@ -0,0 +1,31 @@
+    "columns": [
+	{
+	    "name": "clinical_assertion_trait_set_id",
+	    "datatype": "int",
```
```suggestion
	    "datatype": "integer",
```

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTraitSet.json:
```diff
@@ -0,0 +1,31 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "clinical_assertion_id",
```
`links`?

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionTraitSet.json:
```diff
@@ -0,0 +1,31 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "variation_archive_trait_set_id",
```
`links`?

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionVariation.json:
```diff
@@ -0,0 +1,35 @@
+    "name": "clinical_assertion_variation",
+    "columns": [
+	{
+	    "name": "clinical_assertion_variation_id",
```
```suggestion
	    "name": "id",
```

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionVariation.json:
```diff
@@ -0,0 +1,35 @@
+    "columns": [
+	{
+	    "name": "clinical_assertion_variation_id",
+	    "datatype": "int",
```
```suggestion
	    "datatype": "integer",
```

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionVariation.json:
```diff
@@ -0,0 +1,35 @@
+	},
+	{
+	    "name": "parent_id",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/ClinicalAssertionVariation.json:
```diff
@@ -0,0 +1,35 @@
+	},
+	{
+	    "name": "clinical_assertion_id",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/RCVAccession.json:
```diff
@@ -0,0 +1,63 @@
+    "name": "rcv_accession",
+    "columns": [
+	{
+	    "name": "rcv_accession_id",
```
```suggestion
	    "name": "id",
```

danxmoran commented on schemas/clinvar/tables/RCVAccession.json:
```diff
@@ -0,0 +1,63 @@
+	},
+	{
+	    "name": "submission_count",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/RCVAccession.json:
```diff
@@ -0,0 +1,63 @@
+	},
+	{
+	    "name": "independent_observations",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/RCVAccession.json:
```diff
@@ -0,0 +1,63 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "variation_archive_id",
```
`links`?

danxmoran commented on schemas/clinvar/tables/Submission.json:
```diff
@@ -0,0 +1,22 @@
+			 "datatype": "int"
+		}
+    ]
+}
```
Weird formatting in this file?

danxmoran commented on schemas/clinvar/tables/Submission.json:
```diff
@@ -0,0 +1,22 @@
+		},
+		{
+			"name": "clinical_assertion_id",
+			"datatype": "string"
```
`links`?

danxmoran commented on schemas/clinvar/tables/Submission.json:
```diff
@@ -0,0 +1,22 @@
+			"datatype": "string"
+		},
+		{
+			 "name": "submitter_id",
```
`links`?

danxmoran commented on schemas/clinvar/tables/Submission.json:
```diff
@@ -0,0 +1,22 @@
+		},
+		{
+			 "name": "submitter_id",
+			 "datatype": "int"
```
```suggestion
			 "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/Submitter.json:
```diff
@@ -0,0 +1,31 @@
+    "name": "submitter",
+    "columns": [
+	{
+	    "name": "org_id",
```
```suggestion
	    "name": "id",
```

danxmoran commented on schemas/clinvar/tables/Submitter.json:
```diff
@@ -0,0 +1,31 @@
+    "columns": [
+	{
+	    "name": "org_id",
+	    "datatype": "int",
```
```suggestion
	    "datatype": "integer",
```

danxmoran commented on schemas/clinvar/tables/Submitter.json:
```diff
@@ -0,0 +1,31 @@
+	},
+	{
+	    "name": "clinical_assertion_id",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/Submitter.json:
```diff
@@ -0,0 +1,31 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "clinical_assertion_id",
```
`links`?

danxmoran commented on schemas/clinvar/tables/VariationArchive.json:
```diff
@@ -0,0 +1,71 @@
+    "name": "variation_archive",
+    "columns": [
+	{
+	    "name": "variant_archive_id",
```
```suggestion
	    "name": "id",
```

danxmoran commented on schemas/clinvar/tables/VariationArchive.json:
```diff
@@ -0,0 +1,71 @@
+	},
+	{
+	    "name": "version",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/VariationArchive.json:
```diff
@@ -0,0 +1,71 @@
+	},
+	{
+	    "name": "num_submissions",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/VariationArchive.json:
```diff
@@ -0,0 +1,71 @@
+	},
+	{
+	    "name": "num_submitters",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveRelease.json:
```diff
@@ -0,0 +1,15 @@
+	    ]
+	}
+    ]
+}
```
Should there be more columns in this table? If not, we should drop it since all the info is fully contained in the `variation_archive` table.

danxmoran commented on schemas/clinvar/tables/VariationArchiveTrait.json:
```diff
@@ -0,0 +1,39 @@
+    "name": "variation_archive_trait",
+    "columns": [
+	{
+	    "name": "variation_archive_trait_id",
```
```suggestion
	    "name": "id",
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveTrait.json:
```diff
@@ -0,0 +1,39 @@
+    "columns": [
+	{
+	    "name": "variation_archive_trait_id",
+	    "datatype": "int",
```
```suggestion
	    "datatype": "integer",
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveTrait.json:
```diff
@@ -0,0 +1,39 @@
+		    "table_name": "clinical_assertion_trait",
+		    "column_name": "variation_archive_trait_id"
+		}
+		]
```
```suggestion
	]
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveTrait.json:
```diff
@@ -0,0 +1,39 @@
+	},
+	{
+	    "name": "variation_archive_trait_set_id",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveTrait.json:
```diff
@@ -0,0 +1,39 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "variation_archive_trait_set_id",
```
`links`?

danxmoran commented on schemas/clinvar/tables/VariationArchiveTraitSet.json:
```diff
@@ -0,0 +1,39 @@
+    "name": "variation_archive_trait_set",
+    "columns": [
+	{
+	    "name": "variation_archive_trait_set_id",
```
```suggestion
	    "name": "id",
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveTraitSet.json:
```diff
@@ -0,0 +1,39 @@
+    "columns": [
+	{
+	    "name": "variation_archive_trait_set_id",
+	    "datatype": "int",
```
```suggestion
	    "datatype": "integer",
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveTraitSet.json:
```diff
@@ -0,0 +1,39 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "variation_archive_id",
```
`links`?

danxmoran commented on schemas/clinvar/tables/VariationArchiveTraitSet.json:
```diff
@@ -0,0 +1,39 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "rcv_accession_id",
```
`links`?

danxmoran commented on schemas/clinvar/tables/VariationArchiveVariation.json:
```diff
@@ -0,0 +1,60 @@
+    "name": "variation_archive_variation",
+    "columns": [
+	{
+	    "name": "variation_archive_variation_id",
```
```suggestion
	    "name": "id",
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveVariation.json:
```diff
@@ -0,0 +1,60 @@
+    "columns": [
+	{
+	    "name": "variation_archive_variation_id",
+	    "datatype": "int",
```
```suggestion
	    "datatype": "integer",
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveVariation.json:
```diff
@@ -0,0 +1,60 @@
+	},
+	{
+	    "name": "allele_id",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveVariation.json:
```diff
@@ -0,0 +1,60 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "allele_id",
```
Maybe this is a piece that hasn't been unrolled yet, but should this just be `allele`? I don't see an `allele` table that this ID would reference.

toneillbroad replied:
No, this is just an internally generated but public facing numeric allele identifier from ClinVar. There is no table to be referenced. But I'd prefer to keep this as is.

danxmoran commented on schemas/clinvar/tables/VariationArchiveVariation.json:
```diff
@@ -0,0 +1,60 @@
+	},
+	{
+	    "name": "num_copies",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveVariation.json:
```diff
@@ -0,0 +1,60 @@
+	},
+	{
+	    "name": "parent_id",
+	    "datatype": "int"
```
```suggestion
	    "datatype": "integer"
```

danxmoran commented on schemas/clinvar/tables/VariationArchiveVariation.json:
```diff
@@ -0,0 +1,60 @@
+	    "datatype": "int"
+	},
+	{
+	    "name": "variation_archive_id",
```
`links`?

benjamincarlin replied:
According to @thathert this should be named something other than relationship or references as previously suggest to not be confused with the relationships field in the final JSON. It was suggest to used links. Please let me @danxmoran if I should change this.

danxmoran commented on schemas/clinvar/tables/VariationArchiveVariation.json:
```diff
@@ -0,0 +1,60 @@
+	    "datatype": "string"
+	},
+	{
+	    "name": "rcv_accession_id",
```
`links`?

danxmoran reviewed (commented):
Most comments follow the same theme; I left suggestions that can be auto-applied where it was easy. The biggest-picture comment I have is: 

For `from` -> `to` relationships between tables, it looks like the convention here is to add `links` declarations to the `to` column. I'd prefer the reverse, so that links are declared along with the `from` column. I think it'll make it easier to read and understand the pointers between tables. The downside is that renaming a `to` column will have ripple effects across all the other table JSON files. I think the risk to that downside will effectively go away after we standardize on the convention of naming primary ID columns `id` in each table.

toneillbroad commented:
Dan, regarding your proposal of moving the `links` field from the primary id of the `from` table to the foreign key field of the `to` table, I'm ok with that, but that changes the nature of the field from a `links` json array to a singular `link` object containing `table_name` and `column_name` correct? Are you suggesting that if all primary key fields are named `id` that the new `link` object on the foreign key would simply contain `table_name`? My vote would be for the former vs. the latter.

toneillbroad committed:
Removed tables[] and follow[] from assets.
Removed previously generated clinvar_schame.json.
Removed VariationArchiveRelease.json from assets and tables.
Changed all int to integer.
Moved links to foreign key side of relationships.

danxmoran reviewed (approved):
Thanks @toneillbroad!
